src/main/java/core/basesyntax/strategy/impl/PurchaseFruitTransactionOperationHandler.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseFruitTransactionOperationHandler
+        extends AbstractFruitTransactionOperationHandler {
+    public PurchaseFruitTransactionOperationHandler(FruitDao fruitDao) {
+        super(fruitDao);
+    }
+
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        fruitDao.updateQuantity(fruitTransaction.getFruit(), -fruitTransaction.getQuantity());
```

Comment:
To increase readability better have two methods in fruitDao
addQuantity
substractQuantity